lib/controller.js

```diff
@@ -460,150 +460,121 @@ controller.connectToNetwork = function(opts) {
     });
 };
 
+
+
 controller.update = function(opts) {
   return new Promise(function(resolve, reject) {
-    // If the user requested a list of builds
-    if (opts.list) {
-      // Fetch those builds from the build server
-      logs.info('Fetching available builds...');
-      return updates.fetchBuildList()
-        .then(function(builds) {
-          logs.info('Latest builds:');
-          var maxBuildsListed = 10;
-          var SHAs = Object.keys(builds);
-
-          // Limited the number of builds we print out
-          SHAs = SHAs.splice(0, SHAs.length > maxBuildsListed ? maxBuildsListed : SHAs.length);
-          // Print out details for each build
-          for (var i = 0; i < SHAs.length; i++) {
-            // Grab the specific build
-            var build = builds[SHAs[i]];
-            // Format the release time as a date
-            var released = new Date(build.released);
-            // Print when it was published
-            logs.basic('\t Version:', build.version, '\tPublished:', released.toLocaleString());
-          }
-          // Finish up
-          return resolve();
-        }, reject);
-      // This is a standard update
-    } else {
-      // Fetch the latest version
-      var requestedVersion = 'latest';
-      // Unless a specific build was request
-      if (opts.build) {
-        // Then we'll fetch that build
-        requestedVersion = opts.build;
-      }
+    return updates.requestBuildList().then(function(builds) {
+
+      // tessel update -l
+      if (opts.list) {
+        logs.info('Latest builds:');
 
-      // Grab our Tessel
-      var tesselPromise = Tessel.get(opts);
-      var shaPromise;
-
-      // If we aren't forcing, we'll want to get the current SHA on Tessel 
-      if (!opts.force) {
-        // Once we have the Tessel
-        shaPromise = tesselPromise.then(function(tessel) {
-          // Figure out what commit SHA is running on it
-          return tessel.fetchCurrentBuildInfo()
-            // Once we have the current SHA
-            .then(function(currentSHA) {
-              // Figure out the human readable version for that code release
-              return updates.fetchBuildVersionforSHA(currentSHA);
-
-            });
+        builds.slice(-10).forEach(function(build) {
```

Comment:
These builds are in reverse order when I print them out. Perhaps we should sort them in `requestBuildList`?